### PR TITLE
Rewrite ans processor with SDK

### PR DESCRIPTION
**Pull request by dermanyang**
## Purpose
- Rewrite ANS processor using the sdk
- add new, _non-default_ ProcessorConfig called AnsProcessorConfig that contains additional fields to the default one. Namely, `ans_v1_name_records_table_handle`, `ans_v1_primary_names_table_handle`, and `ans_v2_contract_address`

## Testing
image

### CODE REVIEW

File: rust/sdk-processor/src/processors/ans_processor.rs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+};
+use processor::worker::TableFlags;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use tracing::{debug, info};
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct AnsProcessorConfig {
+    // #[serde(flatten)]
+    // pub default: DefaultProcessorConfig,
```
Comment (dermanyang) [inline rust/sdk-processor/src/processors/ans_processor.rs:35]: Is there a better way to do this? The serde flatten wasn't working out for whatever reason @rtso @yuunlimm

File: rust/sdk-processor/src/processors/ans_processor.rs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            .await?
+            .get_chain_id()
+            .await?;
+        check_or_update_chain_id(grpc_chain_id as i64, self.db_pool.clone()).await?;
+
+        let processor_config = match self.config.processor_config.clone() {
+            ProcessorConfig::AnsProcessor(processor_config) => processor_config,
+            _ => {
+                return Err(anyhow::anyhow!(
+                    "Invalid processor config for Account Transactions Processor: {:?}",
```
Comment (rtso) [inline rust/sdk-processor/src/processors/ans_processor.rs:102]: nit: update error msg

**Review by rtso — APPROVED** [review]

File: rust/sdk-processor/src/steps/ans_processor/ans_extractor.rs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            .contains(TableFlags::ANS_PRIMARY_NAME)
+        {
+            all_ans_primary_names.clear();
+        }
+        if self
+            .deprecated_table_flags
+            .contains(TableFlags::ANS_PRIMARY_NAME_V2)
+        {
+            all_ans_primary_names_v2.clear();
+        }
```
Comment (yuunlimm) [inline rust/sdk-processor/src/steps/ans_processor/ans_extractor.rs:112]: this is deprecated, we can remove parsing `ANS_PRIMARY_NAME_V2`

Comment (dermanyang) [reply rust/sdk-processor/src/steps/ans_processor/ans_extractor.rs:112]: thanks, removed

File: rust/sdk-processor/src/steps/ans_processor/ans_extractor.rs [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    > {
+        let (
+            mut all_current_ans_lookups,
+            mut all_ans_lookups,
+            mut all_current_ans_primary_names,
+            mut all_ans_primary_names,
+            all_current_ans_lookups_v2,
+            all_ans_lookups_v2,
+            all_current_ans_primary_names_v2,
+            _, // AnsPrimaryNameV2 is deprecated.
```
Comment (yuunlimm) [inline rust/sdk-processor/src/steps/ans_processor/ans_extractor.rs:89]: should we also update the parsing logic as a part of migration?

Comment (yuunlimm) [reply rust/sdk-processor/src/steps/ans_processor/ans_extractor.rs:89]: or maybe we could probably revisit once we have all those deprecated tables are migrated over to parquet?

Comment (dermanyang) [reply rust/sdk-processor/src/steps/ans_processor/ans_extractor.rs:89]: I think we should update the parsing logic when we do correctness testing/validation and/or when we delete the old processors and move the parsing code to the SDK directory.

Do any ecosystem devs use this table?

Comment (yuunlimm) [reply rust/sdk-processor/src/steps/ans_processor/ans_extractor.rs:89]: some of the tables are being used, and some are not. + 1 when we do correctness testing and fully migrate them

**Pull request merged**